buildSrc/src/main/kotlin/KotlinConfiguration.kt

```diff
@@ -83,3 +82,49 @@ fun getOverridingKotlinApiVersion(project: Project): String? {
  */
 fun irValidationMode(project: Project): String? =
     project.providers.gradleProperty("kotlin_ir_validation_mode").orNull
+
+/**
+ * Unconditional compiler flags required of Kotlin User Projects. See KT-75078.
+ */
+fun KotlinCommonCompilerOptions.addKotlinUserProjectFlags() {
+    freeCompilerArgs.addAll(
+        "-Xreport-all-warnings",
+        "-Xrender-internal-diagnostic-names",
+    )
+}
+
+/**
+ * Use `kotlin_Werror_override` property to override the state of -Werror:
+ * `true` means that warnings should be treated as errors,`false` means that they should not.
+ */
+private fun warningsAreErrorsOverride(project: Project): Boolean? =
+    when (val prop = project.rootProject.properties["kotlin_Werror_override"] as? String) {
+        null -> null
+        "enable" -> true
+        "disable" -> false
+        else -> error("Unknown value for 'kotlin_Werror_override': $prop")
+    }
+
+/**
+ * Set warnings as errors but allow the Kotlin User Project configuration to override. See KT-75078.
+ */
+fun KotlinCommonCompilerOptions.setWarningsAsErrors(project: Project) {
+    if (warningsAreErrorsOverride(project) == false) {
+        freeCompilerArgs.addAll("-Wextra", "-Xuse-fir-experimental-checkers")
+    } else {
+        allWarningsAsErrors.set(true)
+    }
+}
+
+/**
+ * Pass additional CLI options to the Kotlin compiler.
+ */
+fun KotlinCommonCompilerOptions.addExtraCompilerFlags(project: Project) {
+    val extraOptions = project.rootProject.properties["kotlin_additional_cli_options"] as? String
+    if (extraOptions != null) {
+        LOGGER.info("""Adding extra compiler flags '$extraOptions' for a compilation in the project $${project.name}""")
```

Comment:
It makes sense to use the same logging format across all our projects, and the one from `kotlinx-serialization` looks fine: https://github.com/Kotlin/kotlinx.serialization/pull/2996/files#diff-09e3a69c8143d7af22e62f518f556102bbdea00485f9e30be972303bc6c7dc19R58

Reply:
Thank you) I've fixed that